vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SearchManager.java

```diff
@@ -0,0 +1,267 @@
+        
+        int totalSpacePage = spacePage.getTotalPages();
+        /* if page>total pages,last page is returned */
+        if (page > totalSpacePage) {
+            totalSpacePage = totalSpacePage == 0 ? 1 : totalSpacePage;
+            requestedPageForSpace = PageRequest.of(totalSpacePage - 1, pageSize);   
+            spacePage = spaceSearch(requestedPageForSpace,  searchTerm);
+        }
+        return spacePage;
+    }
+    
+    /**
+     * Method to return the requested modules whose name or description contains the
+     * search string
+     * 
+     * @param searchTerm string which has been searched
+     * @param page.      The page number starts from 1.
+     * @return set of modules whose name or description contains the search string
+     *         in the requested page.
+     */
+    @Override
+    public Page<IModule> searchInModules(String searchTerm, int page) {
+        /* if page<1, 1st page is returned */
+        if (page < 1) {
+            page = 1;
+        }
+        Pageable requestedPageForModule = PageRequest.of(page - 1, pageSize);
+        Page<IModule> modulePage = moduleManager.findByNameOrDescription(requestedPageForModule, searchTerm);
+        int totalModulePage = modulePage.getTotalPages();
+        /*
+         * spring will just return an empty dataset if a page that is greater than the
+         * total number of pages is returned. So, the page given (except when it's < 1,
+         * then set it to 1) and then checking if the total page count is smaller than
+         * the given one (if so, make another request). In most cases the page number
+         * will be within the range of pages, so we would need only one db call. Only in
+         * cases where the page number is wrong, would a second one be needed.if
+         * page>total pages,last page is returned.
+         */
+        if (page > totalModulePage) {
+            totalModulePage = totalModulePage == 0 ? 1 : totalModulePage;
+            requestedPageForModule = PageRequest.of(totalModulePage - 1, pageSize);
+            modulePage = moduleManager.findByNameOrDescription(requestedPageForModule, searchTerm);
+        }
+        return modulePage;
+    }
+
+    /**
+     * Method to return the requested slides whose name or description contains the
+     * search string
+     * 
+     * @param searchTerm string which has been searched
+     * @param page.      The page number starts from 1.
+     * @return set of slides whose name or description contains the search string in
+     *         the requested page.
+     */
+    @Override
+    public Page<ISlide> searchInSlides(String searchTerm, int page) {
+        /* if page<1, 1st page is returned */
+        if (page < 1) {
+            page = 1;
+        }
+        Pageable requestedPageForSlide = PageRequest.of(page - 1, pageSize);
+        Page<ISlide> slidePage = slideManager.findByNameOrDescription(requestedPageForSlide, searchTerm);
```

Comment:
similar to my above comment. If a slide is not in a sequence reachable from the start sequence and the module is not linked, it should not show up in the search results.